cumulus/zombienet/zombienet-sdk/tests/zombie_ci/elastic_scaling/slot_based_authoring.rs

```diff
@@ -0,0 +1,169 @@
+
+	log::info!("Checking if collator-elastic is up");
+	assert!(wait_node_is_up(collator_elastic, 60u64).await.is_ok());
+
+	log::info!("Checking if collator-single-core is up");
+	assert!(wait_node_is_up(collator_single_core, 60u64).await.is_ok());
+
+	log::info!("Assigning cores for the parachain");
+	let assign_cores_call = create_assign_core_call(&[(0, PARA_ID_1), (1, PARA_ID_1)]);
+
+	let relay_client: OnlineClient<PolkadotConfig> = alice.wait_client().await?;
+	let res = submit_extrinsic_and_wait_for_finalization_success_with_timeout(
+		&relay_client,
+		&assign_cores_call,
+		&dev::alice(),
+		60u64,
+	)
+	.await;
+	assert!(res.is_ok(), "Extrinsic failed to finalize: {:?}", res.unwrap_err());
+	log::info!("2 more cores assigned to the parachain");
+
+	for (node, block_cnt) in [(collator_single_core, 20.0), (collator_elastic, 40.0)] {
+		log::info!("Checking block production for {}", node.name());
+		assert!(node
+			.wait_metric_with_timeout(BEST_BLOCK_METRIC, |b| b >= block_cnt, 225u64)
+			.await
+			.is_ok());
+	}
+
+	// We want to make sure that none of the consensus hook checks fail, even if the chain makes
+	// progress. If below log line occurred 1 or more times then test failed.
+	for node in [collator_elastic, collator_single_core] {
+		log::info!("Ensuring none of the consensus hook checks fail at {}", node.name());
+		let result = node
+			.wait_log_line_count_with_timeout(
+				"set_validation_data inherent needs to be present in every block",
+				false,
+				LogLineCountOptions::no_occurences_within_timeout(Duration::from_secs(10)),
+			)
+			.await?;
+		assert!(result.success(), "Consensus hook failed at {}: {:?}", node.name(), result);
+	}
+
+	log::info!("Test finished successfully");
+	Ok(())
+}
+
+async fn build_network_config() -> Result<NetworkConfig, anyhow::Error> {
+	let images = zombienet_sdk::environment::get_images_from_env();
+	log::info!("Using images: {images:?}");
+
+	// Network setup:
+	// - relaychain nodes:
+	// 	 - alice
+	// 	   - validator
+	// 	 - validator[0-7]
+	// 	   - validator
+	// 	   - synchronize only with alice
+	// - parachain nodes
+	//   - recovery-target
+	//     - full node
+	//   - collator-elastic
+	//     - full node
+	//     - collator which is the only one producing blocks
+	NetworkConfigBuilder::new()
+		.with_relaychain(|r| {
+			let r = r
+				.with_chain("rococo-local")
+				.with_default_command("polkadot")
+				.with_default_image(images.polkadot.as_str())
+				.with_genesis_overrides(json!({
+					"configuration": {
+						"config": {
+							"scheduler_params": {
+								"num_cores": 4,
+								"max_validators_per_core": 1
+							},
+							"approval_voting_params": {
+								"max_approval_coalesce_count": 5
+							}
+						}
+					}
+				}))
+				// Have to set a `with_node` outside of the loop below, so that `r` has the right
+				// type.
+				.with_node(|node| node.with_name("alice").with_args(vec![]));
+
+			(0..8).fold(r, |acc, i| {
```

Comment:
We have `"num_cores": 4` and `"max_validators_per_core": 1`, so we could scale down the number of validators here to 5.

Reply:
Done